src/toxcore/tcp/server/server_ext.rs

```diff
@@ -100,19 +117,19 @@ impl ServerExt for Server {
                 .for_each(move |packet| {
                     debug!("Handle {:?} => {:?}", client_pk, packet);
                     server_c_c.handle_packet(&client_pk, packet)
+                        .map_err(|e| e.into())
                 });
 
             // writer = for each Packet from to_client_rx send it to client
             let writer = to_client_rx
-                .map_err(|()| Error::from(ErrorKind::UnexpectedEof))
+                .map_err(|()| ConnectionError::ClientStreamError.into())
```

Comment:
Maybe use `err_msg` here since this error should not be happening?

Reply:
Usually it is `.map_err(|()| unreachable!("rx can't fail"))` like in https://github.com/jgallagher/tokio-chat-example/blob/master/tokio-chat-server/src/main.rs#L198